nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/cosmos/document/AzureCosmosDocumentConnectionControllerService.java

```diff
@@ -0,0 +1,192 @@
+    .required(true)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+    public static final PropertyDescriptor DB_ACCESS_KEY = new PropertyDescriptor.Builder()
+        .name("Cosmos Document DB Access Key")
+        .displayName("Cosmos Document DB Access Key")
+        .description("Cosmos DB Access Key from Azure Portal (Settings->Keys)")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .sensitive(true)
+        .build();
+
+    public static final PropertyDescriptor CONSISTENCY = new PropertyDescriptor.Builder()
+        .name("Cosmos Consistency Level")
+        .displayName("Cosmos Consistency Level")
+        .description("Cosmos Consistency Level to use")
+        .required(false)
+        .allowableValues(CONSISTENCY_STRONG, CONSISTENCY_BOUNDED_STALENESS, CONSISTENCY_SESSION,
+                CONSISTENCY_CONSISTENT_PREFIX, CONSISTENCY_EVENTUAL)
+        .defaultValue(CONSISTENCY_SESSION)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.uri = context.getProperty(URI).getValue();
+        this.accessKey = context.getProperty(DB_ACCESS_KEY).getValue();
+        final ConsistencyLevel clevel;
+        final String selectedConsistency;
+        if(context.getProperty(CONSISTENCY).isSet()){
+            selectedConsistency = context.getProperty(CONSISTENCY).getValue();
+        }else {
+            selectedConsistency =  CONSISTENCY_SESSION;
+        }
+
+        switch(selectedConsistency) {
+            case CONSISTENCY_STRONG:
+                clevel =  ConsistencyLevel.STRONG;
+                break;
+            case CONSISTENCY_CONSISTENT_PREFIX:
+                clevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                break;
+            case CONSISTENCY_SESSION:
+                clevel = ConsistencyLevel.SESSION;
+                break;
+            case CONSISTENCY_BOUNDED_STALENESS:
+                clevel = ConsistencyLevel.BOUNDED_STALENESS;
+                break;
+            case CONSISTENCY_EVENTUAL:
+                clevel = ConsistencyLevel.EVENTUAL;
+                break;
+            default:
+                clevel = ConsistencyLevel.SESSION;
+        }
+
+        if (cosmosClient != null) {
+            closeClient();
+        }
+        createDocumentClient(uri, accessKey, clevel);
+    }
+
+
+    @OnStopped
+    public final void closeClient() {
+        if (this.cosmosClient != null) {
+            cosmosClient.close();
+            cosmosClient = null;
+        }
+    }
+
+    protected void createDocumentClient(final String uri, final String accessKey, final ConsistencyLevel clevel){
+        try {
+            ConnectionPolicy connectionPolicy = ConnectionPolicy.getDefaultPolicy();
+            this.cosmosClient = new CosmosClientBuilder()
+            .setEndpoint(uri)
```

Comment:
minor nit: Is the indenting off here?

Reply:
Done in AzureCosmosDBConnectionControllerService. Note the change of java file.